### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Everywhere

### Description
This PR fixes changes some minor code changes:
- Code formatting
- Extract version info modal into components
- Add class name for IconButton

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: src/components/landing-layout/footer/language-picker.tsx [hunk 1]
```diff
@@ -41,16 +41,16 @@ const languages = {
...
-                          .find((supportedLanguage) => wantedLanguage.substr(0, 2) === supportedLanguage)
-  }
-  return foundLanguage || ''
+  return (
+    Object.keys(languages)
+          .find((supportedLanguage) => wantedLanguage === supportedLanguage) ??
+    Object.keys(languages)
+          .find((supportedLanguage) => wantedLanguage.substr(0, 2) === supportedLanguage) ??
+    ''
+  )
```
Comment (DerMolly) [inline src/components/landing-layout/footer/language-picker.tsx:50]: I'd suggest adding some comments here. It's not the easiest to read…

Comment (mrdrogdrog) [reply src/components/landing-layout/footer/language-picker.tsx]: Added jsdoc

**Review by DerMolly — APPROVED** [review]

File: src/components/landing-layout/footer/language-picker.tsx [hunk 2]
```diff
@@ -40,17 +40,26 @@ const languages = {
...
+ * @param wantedLanguage an ISO 639-1 standard language code
+ */
+const findLanguageCode = (wantedLanguage: string): string => (
+  (
+    Object.keys(languages)
+          .find((supportedLanguage) => wantedLanguage === supportedLanguage)
+  ) ?? (
+    Object.keys(languages)
+          .find((supportedLanguage) => wantedLanguage.substr(0, 2) === supportedLanguage)
+  ) ?? ''
```
Comment (ErikMichelson) [inline src/components/landing-layout/footer/language-picker.tsx:59]: Performance thing: Wouldn't it make sense to cache the return value of `Object.keys(languages)` to not calculate that twice?

File: src/components/landing-layout/footer/language-picker.tsx [hunk 3]
```diff
@@ -65,8 +74,7 @@ const LanguagePicker: React.FC = () => {
       size="sm"
       className="mb-2 mx-auto w-auto"
       value={ findLanguageCode(i18n.language) }
-      onChange={ onChangeLang() }
-    >
+      onChange={ onChangeLang() }>
```
Comment (ErikMichelson) [inline src/components/landing-layout/footer/language-picker.tsx:77]: Any reason for the additional brackets?

```suggestion
      onChange={ onChangeLang }>
```

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: src/components/landing-layout/footer/language-picker.tsx (hunk 3 above)
Comment (mrdrogdrog) [reply src/components/landing-layout/footer/language-picker.tsx]: fixed by refactoring

File: src/components/landing-layout/footer/language-picker.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply src/components/landing-layout/footer/language-picker.tsx:59]: I wrapped it into a memo

**Pull request merged**